packages/google_maps_flutter/lib/src/location.dart

```diff
@@ -106,6 +106,23 @@ class LatLngBounds {
     );
   }
 
+  static LatLngBounds fromLatLngList(List<LatLng> list) {
+    assert(list.isNotEmpty);
+    double x0, x1, y0, y1;
+    for (LatLng latLng in list) {
+      if (x0 == null) {
+        x0 = x1 = latLng.latitude;
+        y0 = y1 = latLng.longitude;
+      } else {
+        if (latLng.latitude > x1) x1 = latLng.latitude;
```

Comment:
I think you forgot to handle the case of a latlng list that goes across the anti-meridian.

See Wikipedia's [180th_meridian](https://en.wikipedia.org/wiki/180th_meridian) for more info.